node/node.d.ts

```diff
@@ -2189,63 +2201,86 @@ declare module "crypto" {
     export interface Signer extends NodeJS.WritableStream {
-        update(data: any): void;
-        sign(private_key: string): Buffer;
-        sign(private_key: string, output_format:  'latin1' | 'hex' | 'base64'): string;
+        update(data: string | Buffer): Signer;
+        update(data: string | Buffer, input_encoding: "utf8" | "ascii" | "latin1"): Signer;
+        sign(private_key: string | { key: string; passphrase: string }): Buffer;
+        sign(private_key: string | { key: string; passphrase: string }, output_format: "latin1" | "hex" | "base64"): string;
     }
     export function createVerify(algorith: string): Verify;
     export interface Verify extends NodeJS.WritableStream {
-        update(data: any): void;
+        update(data: string | Buffer): Verify;
+        update(data: string | Buffer, input_encoding: "utf8" | "ascii" | "latin1"): Verify;
         verify(object: string, signature: Buffer): boolean;
-        verify(object: string, signature: string, signature_format: 'latin1' | 'hex' | 'base64'): boolean;
+        verify(object: string, signature: string, signature_format: string): boolean;
     }
-    export function createDiffieHellman(prime_length: number): DiffieHellman;
-    export function createDiffieHellman(prime: number, encoding?: string): DiffieHellman;
+    export function createDiffieHellman(prime_length: number, generator?: number): DiffieHellman;
+    export function createDiffieHellman(prime: Buffer): DiffieHellman;
+    export function createDiffieHellman(prime: string, prime_encoding: "latin1" | "hex" | "base64"): DiffieHellman;
+    export function createDiffieHellman(prime: string, prime_encoding: "latin1" | "hex" | "base64", generator: number | Buffer): DiffieHellman;
+    export function createDiffieHellman(prime: string, prime_encoding: "latin1" | "hex" | "base64", generator: string, generator_encoding: "latin1" | "hex" | "base64"): DiffieHellman;
     export interface DiffieHellman {
-        generateKeys(encoding?: string): string;
-        computeSecret(other_public_key: string, input_encoding?: string, output_encoding?: string): string;
-        getPrime(encoding?: string): string;
-        getGenerator(encoding: string): string;
-        getPublicKey(encoding?: string): string;
-        getPrivateKey(encoding?: string): string;
-        setPublicKey(public_key: string, encoding?: string): void;
-        setPrivateKey(public_key: string, encoding?: string): void;
+        generateKeys(): Buffer;
+        generateKeys(encoding: "latin1" | "hex" | "base64"): string;
+        computeSecret(other_public_key: Buffer): Buffer;
+        computeSecret(other_public_key: string, input_encoding: "latin1" | "hex" | "base64"): Buffer;
+        computeSecret(other_public_key: string, input_encoding: "latin1" | "hex" | "base64", output_encoding: "latin1" | "hex" | "base64"): string;
+        getPrime(): Buffer;
+        getPrime(encoding: "latin1" | "hex" | "base64"): string;
```

Comment:
The type `"latin1" | "hex" | "base64"` and the type `"utf8" | "ascii" | "latin1"` are used in a lot of places. Can you make an type alias for each and use that instead? So you won't have typos.
```
type InputEncoding = "latin1" | "hex" | "base64";
... input_encoding: InputEncoding
```